py/trtorch/ptq.py

```diff
@@ -0,0 +1,61 @@
+from typing import List, Dict, Any
+import torch
+import os
+
+import trtorch._C
+from trtorch._compile_spec import _parse_compile_spec
+from trtorch._version import __version__
+from types import FunctionType
+import tensorrt as trt
+
+def get_cache_mode_batch(self):
+    return None
+
+def get_batch_size(self):
+    return self.batch_size
+
+def get_batch(self, names):
+    if self.current_batch_idx + self.batch_size > self.data_loader.dataset.data.shape[0]:
+        return None
+    batch = self.dataset_iterator.next()
+    self.current_batch_idx += self.batch_size
+    print("Calibrating batch: ", self.current_batch_idx)
+    # Treat the first element as input and others as targets.
+    if isinstance(batch, list):
+        batch = batch[0].to(torch.device('cuda:0'))

```

Comment:
This doesnt seem right. Does this work in multigpu? Im not sure what the best way to deal with this is.

Reply:
It would be ideal if we could try to re-consume some of the user data from the CompileSpec but its probably going to be messy. I can remember if you can move all dataloader tensors at once at the beginning. I would be fine having the user set it properly.

Reply:
This is written for my specific example which used off-the-shelf cifar10 dataloader code which returned (`inputs`, `targets`). Ideally, the dataloader is expected to be written by user where it is only supposed to return just `inputs`. 
I plan to remove this 
```
if isinstance(batch, list):
        batch = batch[0].to(torch.device('cuda:0'))
```
and just return batch.to(...).data_ptr(). What do you think ?